observability/test/o11y_tests_manager.py

```diff
@@ -0,0 +1,306 @@
+
+    def start_process_in_background(self,
+                                    job: TestJob,
+                                    queue_key: QueueKey,
+                                    extra_args: List[str] = []) -> subprocess.Popen:
+        sponge_log_dir = TestUtil.get_sponge_log_dir(self.args.job_mode, job.job_name)
+        # Note: the corresponding sponge_log.xml is being written in run_o11y_tests.py itself
+        with open(os.path.join(sponge_log_dir, 'sponge_log.log'), 'a') as out:
+            logger.info('Starting %s, in queue %s' % (job.job_name, queue_key))
+            env = os.environ.copy()
+            env['GRPC_VERBOSITY'] = 'DEBUG'
+            return subprocess.Popen(
+                [os.path.join(os.path.dirname(__file__), 'run_o11y_tests.py'),
+                 '--server_lang', str(job.server_lang),
+                 '--client_lang', str(job.client_lang),
+                 '--job_mode', str(self.args.job_mode),
+                 '--test_case', job.test_case] + extra_args,
+                stdout=out,
+                stderr=out,
+                env=env)
+
+    def start_all_job_queues(self) -> Dict[QueueKey, threading.Thread]:
+        threads: Dict[QueueKey, threading.Thread] = {}
+        max_job_queue_size = 0
+        for queue_key, job_queue in self.job_queues.items():
+            thd = threading.Thread(target=self.run_job_queue, args=(queue_key,))
+            threads[queue_key] = thd
+            thd.start()
+            if len(job_queue.jobs) > max_job_queue_size:
+                max_job_queue_size = len(job_queue.jobs)
+        logger.info('Max job queue size: %d' % max_job_queue_size)
+        return threads
+
+    def run_job_queue(self, queue_key: QueueKey) -> None:
+        job_queue = self.job_queues[queue_key]
+        for job in job_queue.jobs:
+            try:
+                proc = self.start_process_in_background(
+                    job,
+                    queue_key,
+                    extra_args = [
+                        '--gke_resource_identifier', job_queue.gke_resource_identifier,
+                        '--port', str(job.port)
+                    ]
+                )
+                logger.info('Waiting for %s to finish' % job.job_name)
+                proc.wait(timeout=TIMEOUT_SECS_BACKGROUND_PROCESS)
+                if proc.returncode != 0:
+                    self.test_runner.set_exit_status(1)
+                    logger.error('Failed: %s' % job.job_name)
+                else:
+                    logger.info('Passed: %s' % job.job_name)
+            except Exception:
+                self.test_runner.set_exit_status(1)
+                logger.error('Failed: %s' % job.job_name)
+                logger.error(traceback.format_exc())
+
+    def run_all_test_cases(self) -> None:
+        threads = self.start_all_job_queues()
+        for queue_key, thd in threads.items():
+            thd.join()
+            logger.info('Queue %s is done' % queue_key)
+
+# Main
```

Comment:
Nit: include with a block like the following:

```python
if __name__ == "__main__":
    test_runner = TestRunner(parse_args())
    test_runner.run_tests()
```

Reply:
Implemented.